src/gui/KM_GUIMenuOptions.pas

```diff
@@ -402,13 +482,94 @@ procedure TKMMenuOptions.Show;
 end;
 
 
+
+procedure TKMMenuOptions.KeybindClick(Sender: TObject);
+begin
+  if ColumnBox_Options_Keys.TopIndex < 0 then Exit;
+
+  if Sender = Button_Options_Keys then
+    PopUp_Options_Keys.Show;
+
+  if Sender = Button_Options_Keys_OK then
+    PopUp_Options_Keys.Hide;
+
+  if Sender = Button_Options_Special_Keys then
+    PopUp_Options_Special_Keys.Show;
+
+  if Sender = Button_Options_Special_Keys_OK then
+    PopUp_Options_Special_Keys.Hide;
+end;
+
+
+procedure TKMMenuOptions.Keybind_ListKeySave(Key: Word; Shift: TShiftState);
+var
+  aID, I, D: Integer;
+begin
+  aID := ColumnBox_Options_Keys.ItemIndex;
+  D := 0;
+  // Never allow to change secret debug keys.
+  if (aID >= 0) and (aID <= (gResKeys.KeyCount - 4)) then
+  begin
+    for I := 0 to gResKeys.KeyCount do
+      if (Key = gResKeys.Keys[I]) or (Key in [121, 122]) then Inc(D);
+
+    if (D <> 0) then
+      exit
+    else
+      gResKeys.SaveKey(aID, Key);
+  end;
+  LoadKeys;
+end;
+
+
 procedure TKMMenuOptions.BackClick(Sender: TObject);
 begin
-  //Return to MainMenu and restore resolution changes
+  // Return to MainMenu and restore resolution changes
   fMainSettings.SaveSettings;
-
   fOnPageChange(gpMainMenu);
 end;
 
 
+procedure TKMMenuOptions.LoadKeys;
+var
+  I: Integer;
+begin
+  gResKeys.LoadKeys;
+  ColumnBox_Options_Keys.ItemIndex := -1;
+  ColumnBox_Options_Keys.Clear;
+  // Hide the debug keys
```

Comment:
This comment is misleading
it should read
// Add all keys except debug ones (last 3)